src/main/java/io/quarkus/gizmo/ClassCreator.java

```diff
@@ -77,6 +86,10 @@ public ClassCreator(ClassOutput classOutput, String name, String signature, Stri
     }
 
     public MethodCreator getMethodCreator(MethodDescriptor methodDescriptor) {
+        if (this.isInterface() && "<init>".equals(methodDescriptor.getName())) {
```

Comment:
`"<init>"` would definitely deserve a constant... that could be reused by other gizmo classes as well. It's a pity that `org.jboss.jandex.Utils.INIT_METHOD_NAME` is not public.